soh/soh/GameMenuBar.cpp

```diff
@@ -860,7 +862,19 @@ namespace GameMenuBar {
                     ImGui::EndMenu();
                 }
                 UIWidgets::PaddedText("Fix Vanishing Paths", true, false);
-                UIWidgets::EnhancementCombobox("gDirtPathFix", zFightingOptions, 0);
+               
+                if (UIWidgets::EnhancementCombobox("gSceneSpecificDirtPathFix", zFightingOptions, 0) && gPlayState != NULL) {
+                    switch (gPlayState->sceneNum) {
+                        case SCENE_SPOT00:
+                        case SCENE_SPOT04:
+                        case SCENE_SPOT15:
+                            CVarSetInteger("gDirtPathFix", CVarGetInteger("gSceneSpecificDirtPathFix", 0));
+                            break;
+                        default:
+                            CVarClear("gDirtPathFix");
+                            break;
+                    }
```

Comment:
if this was moved out into a method then that could be registered on the scene transition hook and called here too (i'm assuming this code is duplicated so you don't need to transition scenes when changing the dropdown value)

Reply:
I'm not sure what you're suggesting here.

Reply:
i'm suggesting making a function that has this logic in it, so something like
```cpp
void DoDirtPathStuff(int32_t sceneNum) {
    switch (sceneNum) {
        case SCENE_SPOT00:
        case SCENE_SPOT04:
        case SCENE_SPOT15:
            CVarSetInteger("gDirtPathFix", CVarGetInteger("gSceneSpecificDirtPathFix", 0));
            return;
        default:
            CVarClear("gDirtPathFix");
    }
}
```

then you could have
```cpp
if (UIWidgets::EnhancementCombobox("gSceneSpecificDirtPathFix", zFightingOptions, 0) && gPlayState != NULL) {
    DoDirtPathStuff(gPlayState->sceneNum);
}
```

and
```cpp
void RegisterPathFix() {
    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnTransitionEnd>([](int32_t sceneNum) {
        DoDirtPathStuff(sceneNum);
    }
}
```

Reply:
DoDirtPathStuff becomes undefined in gamemenubar

Reply:
You need to either define the function in a header that's included in both spots, or just use externs and declare it

Reply:
Does the extra cvar/code need to exist in the first place? Is it not possible to place the scene specific checks within wherever the dirt path fix is applied?

Reply:
@briaguya-ai this works but it kept yelling at me for trying to use int32_t as it wasn't defined. Is using `int` instead safe? or what should it use instead? 

https://github.com/Patrick12115/Shipwright/pull/4/files

Reply:
> Does the extra cvar/code need to exist in the first place? Is it not possible to place the scene specific checks within wherever the dirt path fix is applied?

Probably, but the fix is done within LUS and I was trying to avoid having to do anything more in LUS

Reply:
> Does the extra cvar/code need to exist in the first place? Is it not possible to place the scene specific checks within wherever the dirt path fix is applied?

yea like pat said the original fix is low level gfx code in LUS, which already feels icky but doesn't have access to higher level zelda stuff https://github.com/Kenix3/libultraship/blob/097ffc7d0e2b1a24a796de6cb299869c0f3030a2/src/graphic/Fast3D/gfx_gx2.cpp#L380

Reply:
> it kept yelling at me for trying to use int32_t as it wasn't defined.

which file did it get mad at you in?

Reply:
I did it by adding it to GameInteractor.h. Not sure if that's better or not, but it was included in both already.

Reply:
it shouldn't go in `GameInteractor`. it's defined in `mods.cpp` so it should be declared in `mods.h`. Also `GameInteractor` is a class so it's strange having extra non-class method declarations just kind of chillin in that file

Reply:
Changes pushed at #2907